utils/estimated_histogram.hh

```diff
@@ -276,16 +286,7 @@ public:
      * @return the mean histogram value (average of bucket offsets, weighted by count)
      */
     int64_t mean() const {
-        auto lastBucket = buckets.size() - 1;
-        int64_t elements = 0;
-        int64_t sum = 0;
-        for (size_t i = 0; i < lastBucket; i++) {
-            long bCount = buckets[i];
-            elements += bCount;
-            sum += bCount * bucket_offsets[i];
-        }
-
-        return ((double) (sum + elements -1)/ elements);
+        return (_count) ? ((double)_sample_sum)/_count : 0;
```

Comment:
Again we get better accuracy, but at the cost of doing more work. It's negligible compared to the work done by searching for the right bucket (which will be eliminated by using fixed-point pseudo-logarithms), but still should be justified.